src/interfaces.ts

```diff
@@ -25,37 +33,185 @@ export interface AggregateVerifierLoginParams extends BaseLoginParams {
+   * Also is useful to resume an existing session.
+   */
+  init(): Promise<void>;
+
+  /**
+   * Login into the SDK in an implicit flow and initialize all relevant components.
+   * @param loginParams - Parameters for Implicit Login.
+   */
+  loginWithOauth(loginParams: OauthLoginParams): Promise<void>;
+
+  /**
+   * Login into the SDK using ID Token based login and initialize all relevant components.
+   * @param idTokenLoginParams - Parameters with ID Token based Login.
+   */
+  loginWithJWT(idTokenLoginParams: IdTokenLoginParams): Promise<void>;
+
+  /**
+   * Handle redirect result after login.
+   */
+  handleRedirectResult(): Promise<void>;
+
+  /**
+   * Second step for login where the user inputs their factor key.
+   * @param factorKey: A BN used for encrypting your Device/ Recovery TSS Key Share. You can generate it using `generateFactorKey()` function or use an existing one.
+   */
+  inputFactorKey(factorKey: BN): Promise<void>;
+
+  /**
+   * Returns the current Factor Key and TssShareType in MPC Core Kit State
+   **/
+  getCurrentFactorKey(): IFactorKey;
+
+  /**
+   * Creates a new factor for authentication.
+   * @param CreateFactorParams - Parameters for creating a new factor.
+   */
+  createFactor(CreateFactorParams: CreateFactorParams): Promise<string>;
+
+  /**
+   * Deletes the factor identified by the given public key, including all
+   * associated metadata.
+   * @param factorPub - The public key of the factor to delete.
+   */
+  deleteFactor(factorPub: TkeyPoint): Promise<void>;
+
+  /**
+   * Logs out the user, terminating the session.
+   */
+  logout(): Promise<void>;
+
+  /**
+   * Get user information provided by the OAuth provider.
+   */
   getUserInfo(): UserInfo;
-  inputBackupShare(shareMnemonic: string): Promise<void>;
-  exportBackupShare(): Promise<string>;
-  addSecurityQuestionShare(question: string, password: string): Promise<void>;
-  changeSecurityQuestionShare(question: string, password: string): Promise<void>;
-  recoverSecurityQuestionShare(question: string, password: string): Promise<void>;
-  deleteSecurityQuestionShare(question: string): Promise<void>;
-  addCustomShare(factorKey: BN, metadata: Record<string, string>): Promise<void>;
-  recoverCustomShare(factorKey: BN): Promise<void>;
+
+  /**
+   * Get information about how the keys of the user is managed according to the information in the metadata server.
+   */
   getKeyDetails(): KeyDetails;
+
+  /**
+   * Commit the changes made to the user's account when in manual sync mode.
+   */
   commitChanges(): Promise<void>;
-  logout(): Promise<void>;
-  CRITICAL_resetAccount(): Promise<void>;
+
+  /**
+   * Import an existing private key into the Web3Auth MPC Infrastructure.
+   */
+  importTssKey(tssKey: string, factorPub: TkeyPoint, newTSSIndex?: TssShareType): Promise<void>;
+
+  /**
+   * Export the user's current TSS MPC account as a private key
+   */
+  _UNSAFE_exportTssKey(): Promise<string>;
```

Comment:
we dont need to expose this function here, since tkey instance is public  we can tell clients to call using that function if they as. We should not publicly advertise this function unless it is asked for.

Reply:
hmm, we should confirm with Zhen about this, he asked @ieow  to add it

Reply:
deleted